### PR TITLE
SDGT-7548: Add convertToBool operation

### PR DESCRIPTION
**Pull request by leowotzak**
Converts numbers (across multiple data types), strings, & arrays to boolean values.

Zero/empty values are considered false.

### CODE REVIEW

File: transform/operations.go [hunk 1]
```diff
@@ -418,6 +418,42 @@ func (c *convertToInt64) transform(raw interface{}) (interface{}, error) {
...
+func (c *convertToBool) init(args map[string]string) error {
+	return nil
+}
+
+func (c *convertToBool) transform(raw interface{}) (interface{}, error) {
+	switch in := raw.(type) {
+	case bool:
+		return in, nil
+	case string:
+		return len(in) != 0, nil
```
Comment (gannett-ggreer) [inline transform/operations.go:435]:
```suggestion
	case string:
		return strconv.ParseBool(in)
```

Comment (leowotzak) [reply transform/operations.go]: https://github.com/GannettDigital/jstransform/pull/144/commits/e184684099081c3ed476a3f93e404cf46a81afff

File: transform/operations_test.go [hunk 2]
```diff
@@ -586,6 +586,75 @@ func TestConvertToInt64(t *testing.T) {
...
 
+func TestConvertToBool(t *testing.T) {
+	tests := []opTests{
+		{
+			description: "Working string case",
+			in:          "1",
+			want:        true,
+		},
+		{
+			description: "Failing string case",
```
Comment (adamhostettler) [inline transform/operations_test.go:597]: Technically this is a "working" case as it properly transforms, so probably better to name it as such.

Comment (leowotzak) [reply transform/operations_test.go:597]: You're right "working" is a poor choice of word, I'll replace Working/Failing with True/False

Comment (leowotzak) [reply transform/operations_test.go]: https://github.com/GannettDigital/jstransform/pull/144/commits/5359590e5db0445ebe5d34025a18f4beeec28321

**Review by adamhostettler — APPROVED** [review]

**Pull request merged**